MusicName

SimonRhook opened the pull request:
# Overview
Usage example to demonstrate music name function.

Updated formatting from PR #324 from trimester 3 2024 - note #324 already has 1 approved review

Splashkit Function: `music_name`

# Files Included
- [x] Title and explanation (.txt)
- [x] C++ code (SplashKit)
- [] C++ code (Beyond SplashKit)
- [x] C# code (top-level statements)
- [x] C# code (Object-Oriented Programming)
- [x] Python code
- [x] .webm

# Usage Example Checks
- [x] Simple, clear demonstration of the function
- [x] Code uses Splashkit functions
- [x] Tested in Chrome and Firefox

cameronhum commented on public/usage-examples/audio/music_name-1-example-oop.cs:
```diff
@@ -0,0 +1,34 @@
+        public static void Main()
+        {
+            // Check if audio is ready to use
+            if(! SplashKit.AudioReady())
```
fix spacing in if statement, statement also needs curly braces

cameronhum commented on public/usage-examples/audio/music_name-1-example-sk.cpp:
```diff
@@ -0,0 +1,28 @@
+int main()
+{
+    // Check if audio is ready to use
+    if(! audio_ready())
```
fix spacing

cameronhum commented on public/usage-examples/audio/music_name-1-example-sk.cpp:
```diff
@@ -0,0 +1,28 @@
+    }
+    close_all_windows();
+    free_music(music);
+
```
remove empty line

cameronhum commented on public/usage-examples/audio/music_name-1-example-top-level.cs:
```diff
@@ -0,0 +1,26 @@
+using static SplashKitSDK.SplashKit;
+
+// Check if audio is ready to use
+if(! AudioReady())
```
fix spacing

cameronhum commented on public/usage-examples/audio/music_name-1-example-oop.cs:
```diff
@@ -0,0 +1,34 @@
+            Window window = SplashKit.OpenWindow("Music File", 600, 600);
+
+            // Main Loop
+            while (! SplashKit.QuitRequested())
```
spacing needs fixed here too

cameronhum commented on public/usage-examples/audio/music_name-1-example-sk.cpp:
```diff
@@ -0,0 +1,28 @@
+    open_window("Music File", 600, 600);
+
+    // Main Loop
+    while (! quit_requested())
```
fix spacing

cameronhum commented on public/usage-examples/audio/music_name-1-example-top-level.cs:
```diff
@@ -0,0 +1,26 @@
+{
+    ProcessEvents();
+
+    window.Clear(Color.White);
```
Id recommend changing the name to ColorWhite() to distance it from OOP standards

cameronhum commented on public/usage-examples/audio/music_name-1-example.py:
```diff
@@ -0,0 +1,25 @@
+from splashkit import *
+
```
fix extra spacing

cameronhum commented on public/usage-examples/audio/music_name-1-example.py:
```diff
@@ -0,0 +1,25 @@
+
+# Main Loop
+while not quit_requested():
+
```
remove this extra line that doesnt match structure of other languages

cameronhum reviewed (changes requested):
Overall good example of the function. This just has some styling issues in which I've commented. This works in the website and all the code works individually. All there functionality match and the code is very similar to each other.
![image](https://github.com/user-attachments/assets/bfed1c7b-09a2-4857-b95c-29304ca5b93c)

Looking at this image however it could have a lot more added to it to show how it can be used in real projects rather than just its usage. Something simple you could do is make it look like a MP3 player screen utilizing fill circles to create a "CD" and rectangles for the background and separating it from the music name. That way it would show the function in a way that demonstrates its usefulness much better

# Peer Review

I've reviewed the MusicName Usage example

## Checks

- [x] All required files are present.
  - [x] Example Title (.txt)
  - [x] C++ code (SplashKit)
  - [x] C# code (top-level statements)
  - [x] C# code (Object-Oriented Programming)
  - [x] Python code
- [x] Code correctly uses Splashkit functions.
- [ ] Code clearly demonstrates the function.
- [ ] All versions maintain the same structure and comments.

## Code Tests done

- [x] C++ SplashKit code ran correctly.
- [ ] C++ Beyond SplashKit code ran correctly.
- [x] C# top level code ran correctly.
- [x] C# OOP code ran correctly.
- [x] Python code ran correctly.

## Website Tests done

- [x] npm run build
- [x] npm run preview

SimonRhook commented:
Thanks for the review,

I've addressed your comments for styling fixes and I have also expanded the program to meet your other requested.

I've added a "Next Track" button and 2 new music files so when the user clicks the button it will cycle between the 2 tracks which I feel better displays the function usefulness.

JulianBurton reviewed (approved):
This is a very effective example of Music_Name. It efficiently and effectively shows the implementation of the function.
The code is consistant and works well for its inteded function. I do like the updated window look and the functionality of changing the audio that is on.

# Checks
 All required files are present.
 Example Title (.txt)
 C++ code (SplashKit)
 C# code (top-level statements)
 C# code (Object-Oriented Programming)
 Python code
 Code correctly uses Splashkit functions.
 Code clearly demonstrates the function.
 All versions maintain the same structure and comments.

# Code Tests done
 C++ SplashKit code ran correctly.
 C++ Beyond SplashKit code ran correctly.
 C# top level code ran correctly.
 C# OOP code ran correctly.
 Python code ran correctly.

# Website Tests done
 npm run build
 npm run preview

cameronhum reviewed (approved):
# Peer Review

I've reviewed the MusicName usage example. I love the new functionality. I apologize for the reviews feeling repetitive. These functions are very similar so have required similar advice. Good job at keeping each one very unique now though with the new updates! It looks great

## Checks

- [x] All required files are present.
  - [x] Example Title (.txt)
  - [x] C++ code (SplashKit)
  - [x] C# code (top-level statements)
  - [x] C# code (Object-Oriented Programming)
  - [x] Python code
- [x] Code correctly uses Splashkit functions.
- [x] Code clearly demonstrates the function.
- [x] All versions maintain the same structure and comments.

## Code Tests done

- [x] C++ SplashKit code ran correctly.
- [x] C++ Beyond SplashKit code ran correctly.
- [x] C# top level code ran correctly.
- [x] C# OOP code ran correctly.
- [x] Python code ran correctly.

## Website Tests done

- [x] npm run build
- [x] npm run preview
